packages/transaction-controller/src/utils/swaps.ts

```diff
@@ -159,29 +160,34 @@ export async function updateSwapsTransaction(
     swaps?.hasApproveTx === false &&
     transactionMeta.simulationFails
   ) {
-    await cancelTransaction(transactionMeta.id);
+    cancelTransaction(transactionMeta.id);
     throw new Error('Simulation failed');
   }
 
   const swapsMeta = swaps?.meta as Partial<TransactionMeta>;
 
-  if (!swapsMeta) {
-    return;
-  }
-
-  if (transactionType === TransactionType.swapApproval) {
-    updateSwapApprovalTransaction(transactionMeta, swapsMeta);
-    controllerHubEmitter('transaction-new-swap-approval', {
-      transactionMeta,
-    });
+  let updatedTransactionMeta = transactionMeta;
+  if (swapsMeta) {
```

Comment:
I'm a big fan of avoiding nesting for readability, could we keep the early return and just return the `updatedTransactionMeta` or `transactionMeta` unchanged?

Reply:
Sure, updated here: [38824a3](https://github.com/MetaMask/core/pull/3827/commits/38824a37e125968b6514f10d959b68c1384c6c3e)